server/src/main/java/org/elasticsearch/rest/action/document/RestIndexAction.java

```diff
@@ -122,6 +123,7 @@ public RestChannelConsumer prepareRequest(final RestRequest request, final NodeC
         indexRequest.versionType(VersionType.fromString(request.param("version_type"), indexRequest.versionType()));
         indexRequest.setIfSeqNo(request.paramAsLong("if_seq_no", indexRequest.ifSeqNo()));
         indexRequest.setIfPrimaryTerm(request.paramAsLong("if_primary_term", indexRequest.ifPrimaryTerm()));
+        indexRequest.setRequireAlias(request.paramAsBoolean(DocWriteRequest.REQUIRE_ALIAS, indexRequest.isRequireAlias()));
```

Comment:
We need to add this in `RestBulkAction` also, and make it top-level rather than only allowing it per-operation